### ISSUE TITLE
Error with pointer to array types

### ISSUE THREAD

**Issue by Luthaf** [michaeljones/breathe#187]
Trying to use Breathe to generate documentation for the following function declaration 
```c
typedef CHRP_TOPOLOGY void;
int chrp_topology_angles(CHRP_TOPOLOGY  * topology, size_t (*data)[3], size_t nangles)
```
Result in the following failure:
```
WARNING: Error when parsing function declaration:
If no return type {
Invalid definition: Expected identifier in nested name, got keyword: int [error at 3]
  int chrp_topology_angles(CHRP_TOPOLOGY  * topology, size_t(*) data, size_t nangles)
  ---^
} else if return type {
Invalid definition: Expecting "," or ")" in parameters_and_qualifiers, got "d". [error at 62]
  int chrp_topology_angles(CHRP_TOPOLOGY  * topology, size_t(*) data, size_t nangles)
  --------------------------------------------------------------^
}
```

Looks like the array part of the specification is dropped somewhere in the process. Any idea on why this happens ?

**Comment by vitaut** [comment]
Good catch. Should be fixed by #188.

**Comment by Luthaf** [comment]
This is effectively fixed. Very nice reactivity, thanks !

**Comment by Luthaf** [comment]
I just ran into this one, with one level more of pointers! The last master gives me this error message:

```
Invalid definition: Expecting "," or ")" in parameters_and_qualifiers, got "d". [error at 57]
  int chfl_frame_velocities(CHFL_FRAME  * frame, float(**) data[3], size_t * size)
```

**Comment by vitaut** [comment]
Does it help if you add

```python
import re
param_decl = re.sub(r'(\(\*+)(\))', r'\1' + param_name + r'\2', param_decl)
```
before line https://github.com/michaeljones/breathe/blob/bb95ce2d0e87180940cb41ceb6d0c9b2ff238c10/breathe/renderer/rst/doxygen/compound.py#L267?

This should handle any number of pointers.

**Comment by jakobandersen** [comment]
I don't think that declaration is valid C++. GCC gives me "expected ‘,’ or ‘...’ before ‘data’". What type is 'data' supposed to be?

**Comment by vitaut** [comment]
@jakobandersen I think it should be `float(**data)[3]`, but Doxygen messes everything up and the code that undoes the damage doesn't handle more than one pointer right now.

**Comment by jakobandersen** [comment]
Ah, that makes sense. Sorry for the noise.

**Comment by Luthaf** [comment]
>   Does it help if you add

This regex do not work, but the following works for me:

```python
import re
param_decl = re.sub(r'(\(\*+)(\))\s?(\w*)', r'\1' + r'\3' + r'\2', param_decl)
```

> I don't think that declaration is valid C++. GCC gives me "expected ‘,’ or ‘...’ before ‘data’". What type is 'data' supposed to be?

Yes, this is the error message, the corresponding code is effectively `float (**data)[3]`.

**Comment by vitaut** [comment]
Fixed by #231.
